### PR TITLE
Jv dao service

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

File: src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    private ManufacturerDaoImpl manufacturerDao;
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return manufacturerDao.create(manufacturer);
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return manufacturerDao.get(id);
```
Comment (JustElvis) [inline src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java:23]:
```suggestion
    public Manufacturer get(Long id) {
        return manufacturerDao.get(id);
```
Read common mistakes. you must return not Otional in ServiceImpl

File: src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private DriverServiceDaoImpl driverServiceDao;
+
+    @Override
+    public Driver create(Driver driver) {
+        return driverServiceDao.create(driver);
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        return driverServiceDao.get(id);
```
Comment (JustElvis) [inline src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java:23]:
```suggestion
    public Drive> get(Long id) {
        return driverServiceDao.get(id);
```
Read common mistakes. you must return not Otional in ServiceImpl

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -11,10 +11,11 @@
 import mate.jdbc.exception.DataProcessingException;
 import mate.jdbc.lib.Dao;
 import mate.jdbc.model.Manufacturer;
+import mate.jdbc.service.ManufacturerService;
 import mate.jdbc.util.ConnectionUtil;
 
 @Dao
-public class ManufacturerDaoImpl implements ManufacturerDao {
+public class ManufacturerDaoImpl implements ManufacturerService {
```
Comment (JustElvis) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:18]: ManufacturerDaoImpl implements ManufacturerDao
You need to create layer Dao and layer Service.
You mix them. Why?

File: src/main/java/mate/jdbc/dao/DriverServiceDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Driver;
+import mate.jdbc.service.DriverService;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class DriverServiceDaoImpl implements DriverService {
```
Comment (JustElvis) [inline src/main/java/mate/jdbc/dao/DriverServiceDaoImpl.java:18]: the same

**Review by JustElvis — CHANGES_REQUESTED** [review]

**Review by JustElvis — APPROVED** [review]

File: src/main/java/mate/jdbc/dao/impl/DriverServiceDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class DriverServiceDaoImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        String query = "INSERT INTO drivers (name, license_number) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/impl/DriverServiceDaoImpl.java:24]:
```suggestion
                 PreparedStatement statement  = connection
                 .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```

File: src/main/java/mate/jdbc/dao/impl/DriverServiceDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            throw new DataProcessingException("Couldn't get driver by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String query = "SELECT * FROM drivers WHERE is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/impl/DriverServiceDaoImpl.java:60]:
```suggestion
                 PreparedStatement statement = connection
                 .prepareStatement(query)) {
```

File: src/main/java/mate/jdbc/dao/impl/DriverServiceDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String query = "UPDATE drivers SET name = ?, license_number = ?"
+                + " WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/impl/DriverServiceDaoImpl.java:79]:
```suggestion
                 PreparedStatement statement = connection
                 .prepareStatement(query)) {
```

**Review by Noct2000 — CHANGES_REQUESTED** [review]

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**